client/lib/components/recent_numbers_graph.dart

```diff
@@ -33,8 +33,36 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
 
     return BarChart(
       BarChartData(
-        axisTitleData: FlAxisTitleData(show: false),
-        borderData: FlBorderData(show: false),
+        axisTitleData: FlAxisTitleData(
+          show: false,
+        ),
+        borderData: FlBorderData(
+          show: false,
+        ),
+        barTouchData: BarTouchData(
+          touchTooltipData: BarTouchTooltipData(
+            fitInsideHorizontally: true,
+            fitInsideVertically: true,
+            getTooltipItem: (barChartGroupData, _, barRodData, index) {
+              final date = DateTime.fromMillisecondsSinceEpoch(
+                  widget.timeseries[index].epochMsec.toInt());
+              final formattedDate = DateFormat('MMMM d, yyyy').format(date);
+
+              final commaRegEx = RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))');
+              final formattedCount =
```

Comment:
Great engineering is a study in elegance and efficiency. In other words, I can understand what the regex does with a lot of effort that shouldn't be needed. Comma separators for large numbers is a common enough activity that there will be a standard library for it. As above, this also has the benefit of getting localization for free too.
```suggestion
              final formattedCount = NumberFormat().format(barRodData.y);
```